tests/test_queue.py

```diff
@@ -476,3 +481,70 @@ def test_ApiJobQueue_stop_results_in_queue_threads_not_running(mocker):
 
         job_queue.main_thread.quit.assert_called_once_with()
         job_queue.download_file_thread.quit.assert_called_once_with()
+
+
+def test_ApiJobQueue_emits_main_queue_updated_signal_when_message_or_reply_download_added(
+    mocker,
+):
+    mock_client = mocker.MagicMock()
+    mock_session_maker = mocker.MagicMock()
+
+    with threads(2) as [main_thread, file_download_thread]:
+        job_queue = ApiJobQueue(mock_client, mock_session_maker, main_thread, file_download_thread)
+
+        job_queue.main_queue.api_client = mocker.MagicMock()
+        job_queue.main_queue.session_maker = mocker.MagicMock()
+
+        message_download_job = MessageDownloadJob("mock", "mock", "mock")
+        reply_download_job = ReplyDownloadJob("mock", "mock", "mock")
+
+        mock_main_queue_updated = mocker.patch.object(
+            job_queue.main_queue.queue, "queue_updated_signal"
+        )
+
+        job_queue.main_thread.isRunning = mocker.MagicMock(return_value=True)
+        job_queue.download_file_thread.isRunning = mocker.MagicMock(return_value=True)
+
+        job_queue.enqueue(message_download_job)
+        job_queue.enqueue(reply_download_job)
+
+        mock_main_queue_updated.emit.callcount == 2
+        assert mock_main_queue_updated.emit.call_args_list[0][0][0] == 1
+        assert mock_main_queue_updated.emit.call_args_list[1][0][0] == 2
```

Comment:
:wave: Hello! I've got a special request for you @zekehuntergreen: when testing Qt signal emissions, can you please use `QSignalSpy` instead of `patch`? I know this is not what the rest of the file does, but consistency is not the best choice in this specific case.

Example:
```py
from PyQt5.QtTest import QSignalSpy

# Only if needed, to have an event loop running to process signals:
from tests.helper import app  # noqa: F401


MAX_SIGNAL_WAITING_TIME = 50  # milliseconds are usually more than enough, default is 5 seconds which is a lot to wait for a failure.

message_sent_emissions = QSignalSpy(phone.message_sent)
assert message_sent_emissions.isValid()  # sanity check

message_sent_emissions.wait(MAX_SIGNAL_WAITING_TIME)  # will proceed as soon as the signal is detected
assert len(message_sent_emissions) = 1
assert message_sent_emissions[0] = ["Alice", "Bob", "Hi Bob!"]  # list of arguments of the first message
```

The spy object behaves like a list as far as `len()` and the bracket notation are concerned. You can access the arguments of the calls as if the object was a list of them: `[list_of_args_of_first_message, list_of_args_of_second_message, etc]`.


There are a few advantages to using [`QSignalSpy`][doc]:
- mocking or patching the object under test is less that ideal in general
- `QSignalSpy` allows to work with the actual signals, even though they are asynchronous
- `QSignalSpy` works with re-emitted signals, no so mocking which fails in a non-descriptive way
- it is the recommended way to assert on signal emissions, and I'm pushing for embracing Qt instead of working around it :slightly_smiling_face: 

Of course I'm happy to help doing the refactoring if you want!

  [doc]: https://doc.qt.io/qt-5.15/qsignalspy.html

Reply:
Thanks so much for the suggestion, @gonzalo-bulnes ! I've replaced the patches with `QSignalSpy` in both tests. That is much neater than patching the signal.